frontend/src/metabase/query_builder/components/DatasetEditor/DatasetEditor.jsx

```diff
@@ -199,7 +199,9 @@ function DatasetEditor(props) {
       return columns;
     }
     return orderedColumns
-      .map(col => columns.find(c => isSameField(c.field_ref, col.fieldRef)))
+      .map(
+        col => columns.find(c => isSameField(c.field_ref, col.fieldRef)) || col,
+      )
```

Comment:
AFAIR `orderedColumns` in this case are coming from viz settings and only contain a column name and a field ref. Not sure if we should really use them here?